### PR TITLE
Add new case vectors_check under win_irq_check

### PR DESCRIPTION
**Pull request by peixiu**
Refer to the developer advice:
reduce some steps and re-tidied as a new case here.

id: 2096487
Signed-off-by: Peixiu Hou <phou@redhat.com>

### CODE REVIEW

**Comment by peixiu** [comment]
python3 ConfigTest.py --guestname=Win10 --platform=x86_64 --clone=no --testcase=win_irq_check.vectors_check

 (1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.win_irq_check.vectors_check.with_vioscsi.q35: STARTED
 (1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.win_irq_check.vectors_check.with_vioscsi.q35: PASS (169.96 s)

@qingwangrh @YongxueHong Could you help to review this patch?

Thanks a lot~
Peixiu

**Review by hereischen — APPROVED** [review]

**Comment by peixiu** [comment]
Wow~ Haotong @hereischen , thank you a lot~

File: qemu/tests/win_irq_check.py [hunk 1]
```diff
@@ -81,6 +82,10 @@ def set_msi_fguest(enable=True):
     session = utils_test.qemu.windrv_check_running_verifier(session, vm,
                                                             test, driver_verifier,
                                                             timeout)
+    if check_vectors:
+        if int(get_vectors_fqtree()) == int(params["num_queues"]) + 3:
+            test.log.info("Device vectors as expected")
+            return
```
Comment (YongxueHong) [inline qemu/tests/win_irq_check.py:88]: Shall we raise the test.fail? `if int(get_vectors_fqtree()) != int(params["num_queues"]) + 3`

**Comment by YongxueHong** [comment]
hi @peixiu 
For the commit message, I suggest not mentioning the `Refer to the developer advice:`,  but converting it to our understanding and then writing them in the message so that let other reviewers can get the point easily.

File: qemu/tests/win_irq_check.py [hunk 2]
```diff
@@ -73,6 +73,7 @@ def set_msi_fguest(enable=True):
     device_name = params["device_name"]
     timeout = int(params.get("login_timeout", 360))
     restore_msi = False
+    check_vectors = params.get("check_vectors", False)
```
[the remark this answers is not in the document]
Comment (peixiu) [reply qemu/tests/win_irq_check.py]: Thanks for remind, I updated the code~

**Comment by peixiu** [comment]
Hi @YongxueHong , 

Thanks for your advice, updated the code, could you help to check it again?

Tested Passed:
python3 ConfigTest.py --guestname=Win10 --platform=x86_64 --clone=no --testcase=win_irq_check.vectors_check
 (1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.win_irq_check.vectors_check.with_vioscsi.q35: STARTED
 (1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.win_irq_check.vectors_check.with_vioscsi.q35: PASS (154.90 s)

Thanks~
Peixiu

**Comment by qingwangrh** [comment]
python ConfigTest.py --guestname=Win2019 --platform=x86_64 --clone=no --testcase=win_irq_check.vectors_check
(1/1) Host_RHEL.m8.u8.product_rhel.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.win_irq_check.vectors_check.with_vioscsi.q35: STARTED
 (1/1) Host_RHEL.m8.u8.product_rhel.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.win_irq_check.vectors_check.with_vioscsi.q35: PASS (372.47 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

LGTM

File: qemu/tests/cfg/win_irq_check.cfg [hunk 3]
```diff
@@ -11,16 +11,36 @@
...
-            driver_verifier += " ndis"
-    device_name = "Red Hat VirtIO Ethernet Adapter"
-    device_type = "virtio-net-pci"
-    device_hwid = '"PCI\VEN_1AF4&DEV_1000" "PCI\VEN_1AF4&DEV_1041"'
     variants:
+        - with_netkvm:
+            driver_name = netkvm
+            driver_verifier = ${driver_name}
+            Win2016, Win2019, Win8..1, Win2012..r2:
+                    driver_verifier += " ndis"
```
[the remark this answers is not in the document]
Comment (peixiu) [reply qemu/tests/cfg/win_irq_check.cfg]: Thanks, updated~

**Review by YongxueHong — APPROVED** [review]
LGTM.

**Review by hereischen — APPROVED** [review]

**Pull request merged**